### PR TITLE
feat(keycodes): add utilities for checking modifier keys

### PR DESCRIPTION
**Pull request by crisbeto**
Based off of the conversation in #13790, these changes add some utilities for dealing with modifier keys on keyboard events.

### CODE REVIEW

**Comment by crisbeto** [comment]
@DevVersion I'm not too sure how the tests should be set up with Bazel since this isn't an `NgModule`.

**Comment by DevVersion** [comment]
@crisbeto Have a look in the `coercion` entry-point. It uses the `ts_library` and `jasmine_node_test`.

**Comment by jelbourn** [comment]
This one is definitely a `feat` for the next minor

**Comment by crisbeto** [comment]
@DevVersion by the looks of it, the `jasmine_node_test` isn't running in a browser environment.

**Comment by crisbeto** [comment]
Sorted out the test issues with help from @DevVersion. @jelbourn @DevVersion can you take another look?

File: src/cdk/keycodes/modifiers.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+type ModifierKey = 'altKey' | 'shiftKey' | 'ctrlKey' | 'metaKey';
```
Comment (mmalerba) [inline src/cdk/keycodes/modifiers.ts]: should we use an enum for this?

Comment (crisbeto) [reply src/cdk/keycodes/modifiers.ts]: We could, but I wanted to keep the consumption as short as possible. With the enum people would need another import and the call would look like `hasModifierKeys(event, ModifierKey.altKey, ModifierKey.shiftKey)` which gets a little long.

Comment (jelbourn) [reply src/cdk/keycodes/modifiers.ts]: Strings also make sense here because they are the real properties on `KeyboardEvent`

**Pull request merged**